sui_core/src/consensus_adapter.rs

```diff
@@ -129,91 +129,91 @@ impl ConsensusAdapter {
         }
 
         // Wait for the consensus to sequence the certificate and assign locks to shared objects.
-        timeout(self.max_delay, receiver)
-            .await
-            .map_err(|e| SuiError::ConsensusConnectionBroken(e.to_string()))?
-            .expect("Channel with consensus listener dropped")
+        match timeout(self.max_delay, receiver).await {
```

Comment:
Can you add an extensive comment on what should happen upon timeout?

Reply:
done